src/control/client/pool.go

```diff
@@ -64,9 +64,15 @@ func (c *connList) PoolCreate(req *PoolCreateReq) (*PoolCreateResp, error) {
 		return nil, err
 	}
 
-	poolUUID, err := uuid.NewRandom()
+	poolUUID, err := uuid.Parse(req.UUID)
+
 	if err != nil {
-		return nil, errors.Wrap(err, "generating pool uuid")
+		if req.UUID == "" {
+			poolUUID, err = uuid.NewRandom()
+		}
+		if err != nil {
```

Comment:
Wouldn't block on it, but it does feel a little weird to do this instead of having a specific message for the "failed to generate uuid" error.